class.jetpack.php

```diff
@@ -7258,4 +7261,107 @@ public static function handle_post_authorization_actions( $activate_sso = false,
 
 		Jetpack::state( 'message', 'authorized' );
 	}
+
+	/**
+	 * Check if Gutenberg editor is available
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return bool
+	 */
+	public static function is_gutenberg_available() {
+		return function_exists( 'register_block_type' );
+	}
+
+	/**
+	 * Load Gutenberg editor blocks.
+	 *
+	 * This section meant for unstable phase of developing Jetpack's
+	 * Gutenberg extensions. If still around after Sep. 15, 2018 then
+	 * please file an issue to remove it; if nobody responds within one
+	 * week then please delete the code.
+	 *
+	 *
+	 * Loading blocks is disabled by default and enabled via filter:
+	 *   add_filter( 'jetpack_gutenberg', '__return_true', 10 );
+	 *
+	 * When enabled, blocks are loaded from CDN by default. To load locally instead:
+	 *   add_filter( 'jetpack_gutenberg_cdn', '__return_false', 10 );
+	 *
+	 * Note that when loaded locally, you need to build the files yourself:
+	 * - _inc/blocks/jetpack-editor.js
+	 * - _inc/blocks/jetpack-editor.css
+	 *
+	 * CDN cache is busted once a day or when Jetpack version changes. To customize it:
+	 *   add_filter( 'jetpack_gutenberg_cdn_cache_buster', function( $version ) { return time(); }, 10, 1 );
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return void
+	 */
+	public static function load_jetpack_gutenberg() {
+		/**
+		 * Filter to turn on loading Gutenberg blocks
+		 *
+		 * @since 6.5.0
+		 *
+		 * @param bool false Whether to load Gutenberg blocks
+		 */
+		if ( ! Jetpack::is_gutenberg_available() || ! apply_filters( 'jetpack_gutenberg', false ) ) {
+			return;
+		}
+
+		/**
+		 * Filter to turn off serving blocks via CDN
+		 *
+		 * @since 6.5.0
+		 *
+		 * @param bool true Whether to load Gutenberg blocks from CDN
+		 */
+		if ( apply_filters( 'jetpack_gutenberg_cdn', true ) ) {
+			$editor_script = 'https://s0.wp.com/wp-content/mu-plugins/jetpack/_inc/blocks/jetpack-editor.js';
+			$editor_style = 'https://s0.wp.com/wp-content/mu-plugins/jetpack/_inc/blocks/jetpack-editor.css';
```

Comment:
Does JP have any way of rotating through `s{0,1,2}`?

Reply:
(Or more generally, to conditionally load code from the CDN or locally, respectively -- much like we're doing here?)

Reply:
I kept things simple for this since its primary use will be for development and testing only.

The closest thing I found was in the Masterbar but in many places around Jetpack we just load via `s0`:

https://github.com/Automattic/jetpack/blob/1f532fb9e5d781c8f55c5669f025dea2453b72c3/modules/masterbar/masterbar.php#L146-L156

We have also have nice stuff like these helpers:

```php
Jetpack::dns_prefetch( array(
	'//s0.wp.com',
	'//s1.wp.com',
	'//s2.wp.com',
) );
```
and
https://github.com/Automattic/jetpack/blob/6a5b8de344946c99ef25cea17e119628acedb27f/class.jetpack.php#L7180-L7199

Reply:
>Or more generally, to conditionally load code from the CDN or locally, respectively -- much like we're doing here

Even if there was, I needed to be able to control these via filters so that they wouldn't load even when in development mode.